Regression 0.23.0: multiple mocks with various params

Following test

```python
import httpx
import respx
from httpx import Response

with respx.mock(using="httpx") as rsps:
    rsps.get("http://example.org/", params={"page": 0}).mock(Response(200))
    rsps.get("http://example.org/", params={"page": 1}).mock(Response(400))

    response_1 = httpx.get("http://example.org/", params={"page": 0})
    assert response_1.status_code == 200
    response_2 = httpx.get("http://example.org/", params={"page": 1})
    assert response_2.status_code == 400
```

runs properly up to respx 0.22.0, but it fails in 0.23.0 with

```
respx.models.AllMockedAssertionError: RESPX: <Request('GET', 'http://example.org/?page=0')> not mocked!
```

I was able to debug to a point that second mock overrides the first mock, even though they mock for a different requests.

--- a/respx/patterns.py
+++ b/respx/patterns.py
@@ -294,17 +294,22 @@ class MultiItemsMixin:
             (
                 key,
                 tuple(
-                    (
-                        ANY
-                        if parse_any and v == str(ANY)
-                        else str(ANY)
-                        if encode_any and v == ANY
-                        else v
-                    )
+                    self._item_value(v, parse_any=parse_any, encode_any=encode_any)
                     for v in value.get_list(key)
                 ),
             )
             for key in sorted(value.keys())
+        )
+
+    def _item_value(
+        self, value: Any, parse_any: bool = False, encode_any: bool = False
+    ) -> Any:
+        return (
+            ANY
+            if parse_any and value == str(ANY)
+            else str(ANY)
+            if encode_any and value is ANY
+            else value
         )
 
     def __hash__(self):
@@ -604,6 +609,15 @@ class Files(MultiItemsMixin, Pattern):
             fileobj = fileobj.encode()
 
         return ((filename, fileobj),)
+
+    def _item_value(
+        self, value: Tuple[Any, Any], parse_any: bool = False, encode_any: bool = False
+    ) -> Tuple[Any, Any]:
+        filename, data = value
+        return (
+            super()._item_value(filename, parse_any=parse_any, encode_any=encode_any),
+            super()._item_value(data, parse_any=parse_any, encode_any=encode_any),
+        )
 
     def clean(self, value: RequestFiles) -> MultiItems:
         if isinstance(value, Mapping):

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -501,6 +501,10 @@ def test_pop():
 )
 async def test_params_match(client, url, params, call_url, call_params):
     respx.get(url, params=params) % dict(content="spam spam")
+
+    # Add an extra competing param but with non-matching value, reproduces #311 issue
+    respx.get(url, params={"foo": "<non-matching>"})
+
     response = await client.get(call_url, params=call_params)
     assert response.text == "spam spam"
 

